src/associations/belongs-to-many/belongs-to-many-association.ts

```diff
@@ -8,28 +8,29 @@ import {ModelClassGetter} from "../../model/shared/model-class-getter";
 import {Association} from "../shared/association";
 import {Sequelize} from "../../sequelize/sequelize/sequelize";
 import {UnionAssociationOptions} from "../shared/union-association-options";
+import {ModelType} from "../../model/model/model";
 
-export class BelongsToManyAssociation extends BaseAssociation {
+export class BelongsToManyAssociation<TCreationAttributes, TModelAttributes> extends BaseAssociation<TCreationAttributes, TModelAttributes> {
 
-  constructor(associatedClassGetter: ModelClassGetter,
-              protected options: BelongsToManyOptions) {
+  constructor(associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
+              protected options: BelongsToManyOptions<TCreationAttributes, TModelAttributes>) {
     super(associatedClassGetter, options);
   }
 
   getAssociation(): Association {
     return Association.BelongsToMany;
   }
 
-  getSequelizeOptions(model: typeof Model,
+  getSequelizeOptions(model: ModelType<TCreationAttributes, TModelAttributes>,
                       sequelize: Sequelize): UnionAssociationOptions {
     const options: OriginBelongsToManyOptions = {...this.options as any};
     const associatedClass = this.getAssociatedClass();
     const throughOptions = this.getThroughOptions(sequelize);
 
     const throughModel = typeof throughOptions === 'object' && typeof throughOptions.model !== "string" ? throughOptions.model : undefined;
     options.through = throughOptions;
-    options.foreignKey = getForeignKeyOptions(model, throughModel, this.options.foreignKey);
-    options.otherKey = getForeignKeyOptions(associatedClass, throughModel, this.options.otherKey);
+    options.foreignKey = getForeignKeyOptions(model, throughModel as any, this.options.foreignKey);
+    options.otherKey = getForeignKeyOptions(associatedClass, throughModel as any, this.options.otherKey);
```

Comment:
Is there is no way to avoid this `as any` type ?

Reply:
@Divlo You can see `any` in few other places in this file, seems like this was badly designed to begin with and it's beyond scope of this PR to fix this. I'm not seeing better fix without major redesign and getting modal attributes to work is way more important. Argument is not affecting return type anyway.

Reply:
Right, maybe we'll need to refactor this code, someday.
I'll approve for the moment.